src/mock/filter.js

```diff
@@ -0,0 +1,44 @@
+const filterNames = [
+  `all`, `overdue`, `today`, `favorites`, `repeating`, `archive`
+];
+
+const generateFilters = (tasks) => {
+  const currentDate = new Date();
+
+  let all = 0;
+  let overdue = 0;
+  let today = 0;
+  let favorites = 0;
+  let repeating = 0;
+  let archive = 0;
+
+  for (let task of tasks) {
+    all++;
+    overdue += task[`dueDate`] instanceof Date && task[`dueDate`] < Date.now() ? 1 : 0;
+    today += task[`dueDate`]
+      && task[`dueDate`].getFullYear() === currentDate.getFullYear()
+      && task[`dueDate`].getMonth() === currentDate.getMonth()
+      && task[`dueDate`].getDate() === currentDate.getDate() ? 1 : 0;
+    favorites += task[`isFavorite`] ? 1 : 0;
+    repeating += Object.values(task.repeatingDays).some(Boolean) ? 1 : 0;
+    archive += task[`isArchive`] ? 1 : 0;
+  }
+
+  let result = {
+    all,
+    overdue,
+    today,
+    favorites,
+    repeating,
+    archive,
+  };
+
+  return filterNames.map((it) => {
+    return {
+      name: it,
+      count: result[it],
+    };
+  });
+};
+
+export {generateFilters};
```

Comment:
Да, очень плохо читается
```suggestion
const filtersNames = [
  `all`,
  `overdue`,
  `today`,
  `favorites`,
  `repeating`,
  `tags`,
  `archive`
];

const getFiltersCount = (filter, tasks) => {
  switch (filter) {
    case `all`:
      return tasks.length;
    case `overdue`:
      return tasks.filter((el) => el.dueDate instanceof Date && el.dueDate < Date.now()).length;
    case `today`:
      return tasks.filter((el) => el.dueDate && el.dueDate.getDate() === new Date().getDate()).length;
    case `favorites`:
      return tasks.filter((el) => el.isFavorite).length;
    case `repeating`:
      return tasks.filter((el) => Object.values(el.repeatingDays).some(Boolean)).length;
    case `tags`:
      return tasks.filter((el) => el.tags.size).length;
    case `archive`:
      return tasks.filter((el) => el.isArchive).length;
    default: return 0;
  }
};

const generateFilters = (tasks) => {
  return filtersNames.map((el) => {
    return {
      title: el,
      count: getFiltersCount(el, tasks)
    };
  });
};

export {generateFilters};
```

Reply:
`Date.getDate()` ведь возвращает дату месяца (от 1 до 31).
![chrome_PTVMIme0G4](https://user-images.githubusercontent.com/26359035/79235554-70ae5200-7e74-11ea-8ad9-b35611220447.png)
Получается, что если сравнивать, к примеру, 1.04.2020 и 1.01.1999 только по `.getDate()`, то они будут будут считаться как "Сегодня"?

```
const aprilDate = new Date(2020, 4, 1);
const januaryDate = new Date(1999, 1, 1);

console.log(aprilDate.getDate() === januaryDate.getDate()); // true
```

Придётся сохранить эту портянку?
```
return tasks.filter((it) => {
    return it.dueDate
        && it.dueDate.getFullYear() === today.getFullYear()
        && it.dueDate.getMonth() === today.getMonth()
        && it.dueDate.getDate() === today.getDate();
}).length;
```

Reply:
> Придётся сохранить эту портянку?

Да, лучше было бы функцию создать конечно